server/svix-server/src/v1/endpoints/endpoint.rs

```diff
@@ -483,6 +537,46 @@ async fn recover_failed_webhooks(
     Ok((StatusCode::ACCEPTED, Json(EmptyResponse {})))
 }
 
+async fn get_endpoint_headers(
+    Extension(ref db): Extension<DatabaseConnection>,
+    Path((_app_id, endp_id)): Path<(ApplicationIdOrUid, EndpointIdOrUid)>,
+    AuthenticatedApplication {
+        permissions: _,
+        app,
+    }: AuthenticatedApplication,
+) -> Result<Json<EndpointHeadersOut>> {
+    let endp = endpoint::Entity::secure_find_by_id_or_uid(app.id, endp_id)
+        .one(db)
+        .await?
+        .ok_or_else(|| HttpError::not_found(None, None))?;
+
+    match endp.headers {
+        Some(h) => Ok(Json(h.into())),
+        None => Ok(Json(EndpointHeadersOut::empty())),
```

Comment:
Just a thought: maybe it's better to implement `Default` rather than have our own custom function (`empty()`)?

Reply:
Fixed.